clang/test/CodeGenSYCL/bool-kernel-argument.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+
+
+template <typename name, typename Func>
```

Comment:
Could you please follow the guidelines from https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#dpc-clang-fe-tests for a FE test that is being added?

Reply:
oh sorry, I made the test via copy past from another. Fixed.

I also added a test the coercing only happens on the kernel argument.